LLama/Control/IGenerationControl.cs

```diff
@@ -0,0 +1,31 @@
+﻿using LLama.Abstractions;
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace LLama.Control
+{
+    /// <summary>
+    /// Control the text generation of LLama Executors.
+    /// </summary>
+    public interface IGenerationControl
+    {
+        /// <summary>
+        /// Use the last output text to determine if the generation should stop.
+        /// </summary>
+        /// <param name="context"></param>
+        /// <param name="inferenceParams"></param>
+        /// <param name="lastOutputText"></param>
+        /// <returns></returns>
+        bool ShouldStopGeneration(LLamaContext context, IInferenceParams inferenceParams, string lastOutputText);
+
+        /// <summary>
```

Comment:
I'm not sure we should have this variant.

Any output tokens are going to be fed through a StreamingTokenDecoder somewhere else. To implement this method you would have to duplicate that work again, just for antiprompt processing!

Reply:
The default implementation of it inside LLamaSharp just takes use of `AntiPromptProcessor`. For users it may not be easy to have a good implementation of this interface without reading of source code. However I think the accessibility should be open to users for flexibility.

TBH I'm wondering if you mean to comment at `DefaultTokenizer`, which uses `StreamingTokenDecoder`?

Reply:
> TBH I'm wondering if you mean to comment at DefaultTokenizer, which uses StreamingTokenDecoder?

I really did mean here 😆 

What I'm getting at is **any** implementation of this method:

```csharp
bool ShouldStopGeneration(LLamaContext context, IInferenceParams inferenceParams, IEnumerable<int> lastOutputIds);
```

_Must_ feed the tokens through a `StreamingTextDecoder` internally (or something equivalent).

So thinking about the entire system you would be detokenizing inside this method (for antiprompting) and also somewhere else (to get the actual output), that's a lot of wasted duplicate work! That's why I'm saying we shouldn't have this overload and it's better to accept the strings directly.

Reply:
Looking at your new PR I've realised that I totally misunderstood how this interface is used, sorry for the confusion!

I _thought_ that `ShouldStopGeneration(tokens)` was an alternative to `ShouldStopGeneration(text)` and internally would be detokenizing. I didn't realise it was intended to process the tokens directly!